Repo Analytics

cgarciae opened the pull request:
# What does this PR do?
Adds a script to fetch and visualize metrics of the Flax repo from Github.

Closes #2108

marcvanzee commented on .github/analytics/issue_activity_since_date.gql:
```diff
@@ -0,0 +1,56 @@
+{
+  repository(owner: "_REPO_OWNER_", name: "_REPO_NAME_") {
```
Could you add some comments to these files explaining what exactly they are querying?

marcvanzee commented on .github/analytics/get_repo_metrics.py:
```diff
@@ -0,0 +1,324 @@
+
+  df_issues = _rolling_window(df_issues, _process_issues)
+
+  # TODO: Analyze PR data
```
This doesn't work yet? Or you just haven't looked at it yet?

marcvanzee reviewed (changes requested):
Very cool! So it seems you can also very easily apply this to other repos, right? I think it would be cool to chat about this during the next sync with HuggingFace and ask them whether they are tracking any metrics.

Also, it may be useful for you to glance over the [Google Python Styleguide](https://google.github.io/styleguide/pyguide.html), which we try to follow as much as possible in our project (although we aren't perfect).

cgarciae commented:
Thanks @marcvanzee!

> So it seems you can also very easily apply this to other repos, right?

Yes, I believe if you just change `google` and `flax` on `main` to any other user + repo it should work.

> Also, it may be useful for you to glance over the [Google Python Styleguide](https://google.github.io/styleguide/pyguide.html), which we try to follow as much as possible in our project (although we aren't perfect).

Yeah, sorry about that, most this code is ported from another project, I will fix it to match this project and remove references to non-existing stuff to avoid confusion.

cgarciae commented:
Added 2 metrics for the PRs and modified the ported code a bit to match our style.

cgarciae commented:
Many of the comments were about the ported code, I added a comment noting the source.
Also added comments explaining a bit what each query does.
Finally, last commit adds 2 metrics for PRs.

marcvanzee reviewed (approved):
Looks good now, thanks a lot!
